app/javascript/controllers/object_reporter.js

```diff
@@ -0,0 +1,83 @@
+import bootstrap from "bootstrap/dist/js/bootstrap";
+import qs from "qs";
+import { Controller } from "@hotwired/stimulus";
+import { TabulatorFull as Tabulator } from "tabulator-tables";
+
+export default class extends Controller {
+  static targets = ["columnSelector"];
+
+  static values = {
+    dataUrl: String,
+    dataUrlParams: Object,
+    downloadUrl: String,
+    columnModel: Array,
+  };
+
+  connect() {
+    this.table = new Tabulator("#objectsTable", {
+      layout: "fitDataTable",
+      height: "70vh",
+      addRowPos: "bottom", //when adding a new row, add it to the bottom of the table
+      movableColumns: true, //allow column order to be changed
+      progressiveLoad: "scroll", //enable progressive loading
+      ajaxURL: this.dataUrlValue,
+      ajaxParams: this.dataUrlParamsValue,
+      columns: this.columnModelValue,
+    });
+  }
+
+  downloadCSV() {
+    const flashContainer = document.getElementsByClassName("flash_messages")[0];
+    const downloadInProgressFlash = `
+      <div id="csv-download-flash" class="alert alert-info">
+        Your CSV report is being generated... Please hold.
+        <a class="btn-close close" data-dismiss="alert" data-bs-dismiss="alert" href="#">&times;</a>
+      </div>
+    `;
+    flashContainer.innerHTML = downloadInProgressFlash;
```

Comment:
Pop up, and later clean up, a flash message so users know the CSV may take a little bit to generate and download (for especially large data sets)